datastream-server/src/test/java/com/linkedin/datastream/server/assignment/TestLoadBasedPartitionAssigner.java

```diff
@@ -0,0 +1,248 @@
+
+    DatastreamGroupPartitionsMetadata metadata = new DatastreamGroupPartitionsMetadata(new DatastreamGroup(
+        Collections.singletonList(ds1)), Arrays.asList("P1", "P2", "P3", "P4"));
+
+    LoadBasedPartitionAssigner assigner = new LoadBasedPartitionAssigner();
+    Map<String, Set<DatastreamTask>> newAssignment =
+        assigner.assignPartitions(throughputInfo, currentAssignment, unassignedPartitions, metadata);
+
+    DatastreamTask task3 = (DatastreamTask) newAssignment.get("instance1").toArray()[0];
+
+    // verify that task in instance1 got the new partition
+    Assert.assertEquals(task3.getPartitionsV2().size(), 3);
+    Assert.assertTrue(task3.getPartitionsV2().contains("P4"));
+  }
+
+  private DatastreamTask createTaskForDatastream(Datastream datastream) {
+    return createTaskForDatastream(datastream, Collections.emptyList());
+  }
+
+  private DatastreamTask createTaskForDatastream(Datastream datastream, List<String> partitions) {
+    DatastreamTaskImpl task = new DatastreamTaskImpl(Collections.singletonList(datastream));
+    task.setPartitionsV2(partitions);
+    ZkAdapter mockAdapter = Mockito.mock(ZkAdapter.class);
+    Mockito.when(mockAdapter.checkIsTaskLocked(anyString(), anyString(), anyString())).thenReturn(true);
+    task.setZkAdapter(mockAdapter);
+    return task;
+  }
+
+  private ClusterThroughputInfo getDummyClusterThroughputInfo(List<String> partitions) {
+    Map<String, PartitionThroughputInfo> partitionThroughputMap = new HashMap<>();
+    for (String partitionName : partitions) {
+      int bytesInRate = 5;
+      int messagesInRate = 5;
+      partitionThroughputMap.put(partitionName,
+          new PartitionThroughputInfo(bytesInRate, messagesInRate, partitionName));
+    }
+    return new ClusterThroughputInfo("dummy", partitionThroughputMap);
+  }
+
+  @Test
+  public void insertionTests() {
+    LoadBasedPartitionAssigner assigner = new LoadBasedPartitionAssigner();
+
+    ArrayList<String> list1 = new ArrayList<>(Arrays.asList("P1", "P2", "P3"));
+    String partitionToInsert1 = "P4";
+    Map<String, Integer> throughputMap1 = new HashMap<>();
+    throughputMap1.put("P1", 0);
+    throughputMap1.put("P2", 0);
+    throughputMap1.put("P3", 0);
+    throughputMap1.put("P4", 0);
+    assigner.insertTaskIntoSortedList(partitionToInsert1, list1, throughputMap1);
+    Assert.assertEquals(list1.size(), 4);
+    Assert.assertTrue(list1.contains(partitionToInsert1));
+
+    ArrayList<String> list2 = new ArrayList<>(Arrays.asList("P1", "P2"));
+    String partitionToInsert2 = "P3";
+    Map<String, Integer> throughputMap2 = new HashMap<>();
+    throughputMap2.put("P1", 0);
+    throughputMap2.put("P2", 1);
+    throughputMap2.put("P3", 2);
+    assigner.insertTaskIntoSortedList(partitionToInsert2, list2, throughputMap2);
+    Assert.assertEquals(list2.size(), 3);
+    Assert.assertTrue(list2.contains(partitionToInsert2));
+    Assert.assertTrue(list2.get(list2.size() - 1).equals(partitionToInsert2));
```

Comment:
nit: simplify `assertTrue` to `assertEquals`.

Here and bunch of other places in this file.